maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/AbstractSurefireMojoProvidersInfoTest.java

```diff
@@ -0,0 +1,172 @@
+
+        // no interaction with artifact only reference are checked
+        verifyNoMoreInteractions( testNg );
+    }
+
+    @Test
+    public void testNgProviderNotApplicable()
+    {
+        ProviderInfo providerInfo = mojo.new TestNgProviderInfo( null );
+        assertThat( providerInfo.isApplicable() ).isFalse();
+    }
+
+    //surefire-junit-platform
+
+    @Test
+    public void jUnitPlatformProviderApplicable()
+    {
+        Artifact junitPlatform = mock( Artifact.class );
+        ProviderInfo providerInfo = mojo.new JUnitPlatformProviderInfo( junitPlatform, aTestClassPath() );
+
+        assertThat( providerInfo.isApplicable() ).isTrue();
+
+        // no interaction with artifact only reference are checked
+        verifyNoMoreInteractions( junitPlatform );
+    }
+
+    @Test
+    public void jUnitPlatformProviderNotApplicable()
+    {
+        ProviderInfo providerInfo = mojo.new JUnitPlatformProviderInfo( null, aTestClassPath() );
+        assertThat( providerInfo.isApplicable() ).isFalse();
+    }
+
+    // surefire-junit4
+
+    @Test
+    public void jUnit4ProviderNullArtifacts()
+    {
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( null, null );
+        assertThat( providerInfo.isApplicable() ).isFalse();
+    }
+
+    @Test
+    public void jUnit4ProviderOnlyJunitDepArtifact()
+    {
+        Artifact junitDep = mock( Artifact.class );
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( null, junitDep );
+
+        assertThat( providerInfo.isApplicable() ).isTrue();
+
+        // no interaction with artifact only reference are checked
+        verifyNoMoreInteractions( junitDep );
+    }
+
+
+    @Test
+    public void jUnit4ProviderJunit3WithJDepArtifact()
+    {
+        Artifact junit = aArtifact( "3.8.1" );
+        Artifact junitDep = mock( Artifact.class );
+
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( junit, junitDep );
+
+        // ??? only existing for junitDep in any version is checked
+        assertThat( providerInfo.isApplicable() ).isTrue();
+
+        // no interaction with artifact only reference are checked
+        verifyNoMoreInteractions( junitDep );
+    }
+
+    @Test
+    public void jUnit4ProviderJunit3AsDependencyArtifact()
+    {
+        Artifact junit = aArtifact( "3.8.1" );
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( junit, null );
+        assertThat( providerInfo.isApplicable() ).isFalse();
+    }
+
+    @Test
+    public void jUnit4ProviderJunit45AsDependencyArtifact()
+    {
+        Artifact junit = aArtifact( "4.5" );
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( junit, null );
+        assertThat( providerInfo.isApplicable() ).isTrue();
+    }
+
+    @Test
+    public void jUnit4ProviderJunit47AsDependencyArtifact()
+    {
+        Artifact junit = aArtifact( "4.7" );
+        ProviderInfo providerInfo = mojo.new JUnit4ProviderInfo( junit, null );
+        // ??? it is ok for 4.7 ???
```

Comment:
This is ok, it is suitable provider info also for the 4.7+ if you did not configure parallel/groups.
But I see that you have missed the `JUnitCoreProviderInfo` - it is for 4.7+ and parallel/groups.

Reply:
There is separate test for it `AbstractSurefireMojoJunitCoreProvidersInfoTest`